src/main/java/pl/coderstrust/accounting/controller/InvoiceController.java

```diff
@@ -153,4 +161,21 @@ public void removeInvoice(@RequestBody int id) {
     invoiceService.updateInvoice(invoice);
     return ResponseEntity.ok().build();
   }
-}
\ No newline at end of file
+
+  @RequestMapping(value = "pdf/{id}", method = RequestMethod.GET, produces = MediaType.APPLICATION_PDF_VALUE)
+  @ApiOperation(value = "Returns invoice in pdf format")
+  public ResponseEntity<byte[]> invoicePdf(@PathVariable("id") int id) {
+    Optional<Invoice> invoiceFromDatabase = invoiceService.findById(id);
+    if (!invoiceFromDatabase.isPresent()) {
+      return ResponseEntity.notFound().build();
+    }
+    try {
+      ByteArrayOutputStream arrayOutputStream = new PdfService().generatePdf(PdfSet.createContent(
+          invoiceFromDatabase.get()), invoiceFromDatabase.get(), invoiceFromDatabase.get().getIdentifier() + ".pdf");
+      return ResponseEntity.ok(arrayOutputStream.toByteArray());
+    } catch (IOException ioex) {
+      ioex.printStackTrace();
+    }
+    return ResponseEntity.ok().build();
```

Comment:
That's not correct - you cannot return ok without pdf content - it should be rather error - please see comment above.